app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileActivity.java

```diff
@@ -0,0 +1,170 @@
+                    .setConfirmText(getString(R.string.dialog_ok))
+                    .setConfirmClickListener(null)
+                    .show();
+        }
+    }
+
+    /**
+     * Generates a KeyPair based on the type selected.
+     *
+     * @return a {@link KeyPair} with the key type selected
+     */
+    private KeyPair generateKeyPair() {
+        final RadioButton rsaKeyButton = (RadioButton) findViewById(R.id.radioRSAKey);
+        if (rsaKeyButton.isChecked()) {
+            return serviceLocator.getKeyGenerator().generateRSAKeyPair("NervousFish generated key");
+        }
+        return null;
+    }
+
+    /**
+     * Validates if the input fields are not empty and if the input is valid.
+     * This also means that the password and the repeat password should be the same,
+     * and the password length is larger or equal to 6.
+     *
+     * @return a {@link boolean} which is true if all fields are valid
+     */
+    private boolean validateInputFields() {
+        boolean allValid = true;
+
+        final EditText nameInputField = (EditText) this.findViewById(R.id.profileEnterName);
+        if (isValidName(nameInputField.getText().toString())) {
+            nameInputField.setBackgroundColor(Color.TRANSPARENT);
+        } else {
+            nameInputField.setBackgroundColor(ResourcesCompat.getColor(getResources(), R.color.red_fail, null));
+            allValid = false;
+        }
+
+        final EditText passwordInputField = (EditText) this.findViewById(R.id.profileEnterPassword);
+        if (isValidName(passwordInputField.getText().toString())
+                && passwordInputField.getText().toString().length() >= MIN_PASSWORD_LENGTH) {
+            passwordInputField.setBackgroundColor(Color.TRANSPARENT);
+        } else {
+            passwordInputField.setBackgroundColor(ResourcesCompat.getColor(getResources(), R.color.red_fail, null));
+            allValid = false;
+        }
+
+        final EditText passwordRepeatInputField = (EditText) this.findViewById(R.id.profileRepeatPassword);
+        if (isValidName(passwordRepeatInputField.getText().toString())) {
+            passwordRepeatInputField.setBackgroundColor(Color.TRANSPARENT);
+        } else {
+            passwordRepeatInputField.setBackgroundColor(ResourcesCompat.getColor(getResources(), R.color.red_fail, null));
+            allValid = false;
+        }
+
+        if (!passwordInputField.getText().toString().equals(passwordRepeatInputField.getText().toString())) {
+            passwordInputField.setBackgroundColor(ResourcesCompat.getColor(getResources(), R.color.red_fail, null));
+            passwordRepeatInputField.setBackgroundColor(ResourcesCompat.getColor(getResources(), R.color.red_fail, null));
+            allValid = false;
+        }
+
+        return allValid;
+    }
+
+    /**
+     * Will return true if the name is valid. This means
+     * that it has at least 1 ASCII character.
+     *
+     * @param name - the name that has been entered
```

Comment:
For the sake of consistency I'd remove the - character after the param name as we mostly do it without. I personally however like this style, but it was frowned upon in my SEM project group.